sync code

lvan100 opened the pull request.

llitfkitfk commented:
> 赞！还可以到 https://github.com/go-spring/go-spring-boot-starter 提交一个开箱即用的模块。

https://github.com/go-spring/go-spring-boot-starter/pull/1

lvan100 committed:
ReadMe 文件增加 Go 版本图标

lvan100 committed:
更新 ReadMe 文件

lvan100 committed:
Merge pull request #42 from lvan100/master

ReadMe 文件增加 Go 版本图标

lvan100 committed:
新增文档《项目贡献指南》

lvan100 committed:
Merge pull request #1 from go-spring/master

项目同步

lvan100 committed:
更新 《项目贡献指南》

lvan100 committed:
Merge pull request #43 from lvan100/master

更新 《项目贡献指南》

lvan100 committed:
对 http.ResponseWriter 接口进行增强，可以返回发送给客户端的 JSON 数据

lvan100 committed:
对 http.ResponseWriter 接口进行增强，可以返回发送给客户端的 JSON 数据

lvan100 committed:
对 http.ResponseWriter 接口进行增强，可以返回发送给客户端的 JSON 数据

lvan100 committed:
对 http.ResponseWriter 接口进行增强，可以返回发送给客户端的 JSON 数据

lvan100 committed:
更新 spring-web 版本

lvan100 committed:
更新 spring-web 版本

lvan100 committed:
同步 spring-error 项目为「已废弃」状态

lvan100 committed:
DefaultLoggerContext 去掉 Context() 函数，容易引起混乱

lvan100 committed:
本次更新的内容较多，详情如下：

1. 重构 WebContext 的 Response 系列函数，均允许抛出 panic；
2. 默认集成 RecoveryFilter 并进行适配，替换 Abort 中断机制；
3. 增加用户自定义错误处理函数，简化默认的 RPC 执行函数实现；
4. 去掉 WebContext 的 Context() 函数，可由 Request 获取。

lvan100 committed:
重构 Response 系列函数；默认集成 RecoveryFilter，替换 Abort 中断机制

lvan100 committed:
重构 Response 系列函数；默认集成 RecoveryFilter

lvan100 committed:
移除 spring-error 项目

lvan100 committed:
更新依赖

lvan100 committed:
更新依赖

lvan100 committed:
更新依赖

lvan100 committed:
增加一个 spring-web rpc 的示例

lvan100 committed:
完善 ErrorHandler 默认实现

lvan100 committed:
完善 RecoveryFilter 处理逻辑

lvan100 committed:
完善 RecoveryFilter 处理逻辑

lvan100 committed:
更新依赖

lvan100 committed:
更新依赖

lvan100 committed:
panic(*rpc.RpcResult) 和 return(*rpc.RpcResult) 等价处理

ymiras committed:
去掉 statusCode，需要设置状态码时，手动调用 WebContext.Status(code int) 进行设置

lvan100 committed:
Merge pull request #47 from acrossmountain/master

去掉 statusCode

lvan100 committed:
去掉 WebContext 响应函数的 StatusCode 参数

lvan100 committed:
去掉 WebContext 响应函数的 StatusCode 参数

lvan100 committed:
去掉 WebContext 响应函数的 StatusCode 参数

lvan100 committed:
更新依赖

lvan100 committed:
更新依赖

lvan100 committed:
更新依赖

lvan100 committed:
修复循环引用问题

lvan100 committed:
更新依赖

lvan100 committed:
测试去掉 WebContext 响应函数的 StatusCode 参数

lvan100 committed:
支持函数定义 Filter 对象；增加 RpcResult 定义 RPC 返回值 (推荐标准)

lvan100 committed:
支持 MethodNotAllowed

lvan100 committed:
更新依赖

lvan100 committed:
更新依赖

lvan100 committed:
修复 panic 发生位置信息错误

lvan100 committed:
测试 spring-gin 支持 MethodNotAllowed

lvan100 committed:
统一 RpcError 错误信息

lvan100 committed:
统一 RpcError 错误信息

lvan100 committed:
更新测试项目到最新依赖

lvan100 committed:
将核心模块的公开函数列出来当做第一版的文档

lvan100 committed:
测试 github 的 markdown 目录语法是否可用

lvan100 committed:
测试 github 的 markdown 目录语法是否可用

lvan100 committed:
测试 github 的 markdown 目录语法是否可用

lvan100 committed:
梳理 spring-logger 的接口和文档

lvan100 committed:
为 spring-logger 项目增加示例说明

lvan100 committed:
重新梳理 spring-logger 架构，简化框架实现

lvan100 committed:
去掉外部依赖

lvan100 committed:
去掉外部依赖；增加 Assert* 系列函数；更新说明文档

lvan100 committed:
测试代码覆盖率插件

lvan100 committed:
去掉外部依赖

lvan100 committed:
增加 CI 配置文件

lvan100 committed:
日志组件和 WebContext 分离；使用 spring-utils 进行单元测试；

lvan100 committed:
移除 spring-test 模块

lvan100 committed:
Squashed 'spring/spring-redoc/' content from commit 9dc8204

git-subtree-dir: spring/spring-redoc
git-subtree-split: 9dc8204b76e6030f3441b23cff8a2202a003864f

lvan100 committed:
Merge commit '124d260b0f37bee3f9921bfcd063beab86b79a5b' as 'spring/spring-redoc'

lvan100 committed:
移除 swagger 处理器实现；WebContainer Start 改为阻塞调用

lvan100 committed:
提交 redoc 实现

lvan100 committed:
重塑 WebServer 的启停接口

lvan100 committed:
更新项目依赖

lvan100 committed:
添加 spring-redoc 项目

lvan100 committed:
添加捐赠收款账户

lvan100 committed:
Create FUNDING.yml

lvan100 committed:
删除 WebServer；WebContainer 增加 BasePath 配置项

lvan100 committed:
适配最新的 spring-web 和 spring-logger 模块

lvan100 committed:
适配最新的 spring-web 和 spring-logger 模块

lvan100 committed:
修改 PatchValue 函数定义

lvan100 committed:
更新依赖包

lvan100 committed:
更新依赖包；去掉 OnPorts 函数

lvan100 committed:
添加详细文档的链接
